articles/reactive-sink.md

```diff
@@ -4,3 +4,8 @@ title: Sink
 ---
 
 ![Marble diagram](~/images/reactive-sink.svg)
+
+The `Sink` operator uses the nested workflow to specify a reactive operation over the elements of the source sequence. `Sink` emits all values from the source sequence directly without modification, and the nested sequence is subscribed to purely for its side-effects. All notifications emitted by the inner [WorkflowOutput](xref:Bonsai.Expressions.WorkflowOutputBuilder) node are ignored.
+
+> [!Tip]
+> `Sink` operators are very useful to prevent side-effects from affecting the source sequence. Even if the inner sequence terminates early, or changes the timing of emitted values, these effects will not propagate to the outer sequence.
```

Comment:
Add warning that if the Sink operation is "assynchronous" (e.g. a Delay(2s)) and the outer sequence finishes before the sink closes, the sink will close.

Reply:
This is a good point, however now I'm divided whether this is something that we want to document, or whether it is an actual issue with `Sink` 🤔

Reply:
I really think this should be the default behavior. I have two arguments:
1. Simplicity and rule generalization. When you close a branch, in all instances that I know of, all the observables created in that branch will also be closed (i.e. SelectMany). It is a very helpful behavior and assurance at the same time.
2. I think users will forget to dispose of the "Sink" and bad things will happen ...

Reply:
Actually, the case with `SelectMany` is a bit more complicated. If you cancel the entire branch from the end, then indeed everything upstream will be cancelled. However, if the input to `SelectMany` emits a single value that starts a 10 min nested computation and then immediately completes, the 10 min computation will still run its course.

In fact, this is a requirement for reactive state machines to work: the next state started off by `SelectMany` should not be cancelled just because the previous state completed.

I guess my question was more on whether `Sink` should be more like `SelectMany` in its side-effects, but I do agree that having runaway computations is a bit unexpected from an operator that is explicitly supposed *not* to allow you to hijack the main sequence.

Maybe the best way to address this for now is to recommend the alternative way to do it, which is actually the best way in many cases: just redirect side-effects to a subject and deal with them elsewhere.

Reply:
Ye you are right, I for some reason I was thinking about the nested behavior not branches...

Still, I think Sink should remain as is. If anything, we could think about a different "Sink" operator that does indeed implement the behavior you are describing.